nir/ir.py

```diff
@@ -23,6 +23,14 @@ class NIRGraph(NIRNode):
 
     nodes: typing.Dict[str, NIRNode]  # List of computational nodes
     edges: Edges
+    
+    @property
+    def inputs(self):
+        return {name:node for name,node in self.nodes.items() if isinstance(node, Input)}
```

Comment:
I get a warning from ruff that the line is too long. May I suggest either adding pre-commit [as mentioned in the docs](https://nnir.readthedocs.io/en/latest/dev.html) or maybe bumping the minimum line length to something more than 88?

Reply:
Or maybe it's just a matter of rebasing onto main?